src/network/serverpackethandler.cpp

```diff
@@ -1358,7 +1358,8 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		Check that target is reasonably close
 		(only when digging or placing things)
 	*/
-	if (action == 0 || action == 2 || action == 3) {
+	if ((!isSingleplayer() && !g_settings->getBool("disable_anticheat")) &&
+			(action == 0 || action == 2 || action == 3)) {
```

Comment:
The action checks should be first in this list because they can be checked much faster than the tree lookup required for `getBool`.